include/CLI/impl/App_inl.hpp

```diff
@@ -285,7 +285,7 @@ CLI11_INLINE Option *App::set_help_flag(std::string flag_name, const std::string
     // Empty name will simply remove the help flag
     if(!flag_name.empty()) {
         help_ptr_ = add_flag(flag_name, help_description);
-        help_ptr_->configurable(false);
+        help_ptr_->configurable(false)->callback_priority(CallbackPriority::PreRequirementsCheck);
```

Comment:
```suggestion
        help_ptr_->configurable(false)->callback_priority(CallbackPriority::PreReadConfig);
```
This will have to change to PreReadConfig to keep the default ordering the same